### PR TITLE
chore: hide progressive onboarding on first session

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [DIA-157] 

### Description

This PR comes as a fix to showing the progressive onboarding on the first app session.

**PS**: The new flag can be useful in many other context in the app

https://github.com/artsy/eigen/assets/11945712/f3c3b909-0890-4f5d-ab06-614a021c87cc

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- hide progressive onboarding on first session - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[DIA-157]: https://artsyproduct.atlassian.net/browse/DIA-157?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (hide progressive onboarding on first session - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0f5411ef3a41a77ebc308421c66274415be7fd03
</p>

File: src/app/store/AuthModel.ts [hunk 1]
```diff
@@ -479,6 +481,10 @@ export const getAuthModel = (): AuthModel => ({
     if (result.status === 201) {
       postEventToProviders(tracks.createdAccount({ signUpMethod: oauthProvider }))
 
+      actions.setSessionState({
+        isFirstSession: true,
+      })
+
```
Comment (olerichter00) [inline src/app/store/AuthModel.ts:487]: So, is the new flag only set to 'true' if the user starts a session by creating an account?

**Review by olerichter00 — APPROVED** [review]
Nice 👍

File: src/app/store/AuthModel.ts (hunk 1 above)
Comment (MounirDhahri) [reply src/app/store/AuthModel.ts:487]: if you signed up from web and switched, we don't cover that yet.

Comment (olerichter00) [reply src/app/store/AuthModel.ts:487]: I guess, that's good enough

**Pull request merged**